src/main/scala/io/iohk/ethereum/ledger/Ledger.scala

```diff
@@ -136,6 +138,30 @@ class LedgerImpl(vm: VM, blockchainConfig: BlockchainConfig) extends Ledger with
         }
   }
 
+  override def simulateTransaction(stx: SignedTransaction, blockHeader: BlockHeader, storages: BlockchainStorages, validators: Validators): TxResult = {
+    val blockchain = BlockchainImpl(storages)
+    val stateRoot = blockchain.getBlockHeaderByHash(blockHeader.hash).map(_.stateRoot)
```

Comment:
Minor detail: I'm wondering, do we need to access the `blockchain` to get the `stateRoot`? Could we just access the `blockHeader` parameter?